src/platform/site-wide/user-nav/selectors.js

```diff
@@ -4,14 +4,21 @@ import { startCase, toLower } from 'lodash';
 import conditionalStorage from '../../utilities/storage/conditionalStorage';
 import { selectProfile } from '../../user/selectors';
 
+const desktopMediaQuery = window.matchMedia('(min-width: 768px)');
+
 export const selectUserGreeting = createSelector(
   state => selectProfile(state).userFullName,
+  state => selectProfile(state).email,
   () => conditionalStorage().getItem('userFirstName'),
-  (name, sessionFirstName) => {
+  (name, email, sessionFirstName) => {
     if (name.first || sessionFirstName) {
       return startCase(toLower(name.first || sessionFirstName));
     }
 
+    if (desktopMediaQuery.matches) {
+      return email;
+    }
+
     return 'My Account';
```

Comment:
I think the CSS responsive utility classes may be a better option here, because with the CSS approach you'll be able to resize the browser window and see the username field respond automatically, whereas to do that in JS, you'll have to set an event listener for a consistent display.

I don't know the responsive utility classes off the top of my head unfortunately, but assuming this function can return JSX without any issues, the markup should be something like -

```jsx
return (
  <div>
    <span className="desktop-only">{email}</span>
    <span className="desktop-hidden">My account</span>
  </div>
);
```

Reply:
Yeah, I agree with this.

Reply:
@jbalboni @ncksllvn I didn't even know we had this available. I will make the updates.

Reply:
@ncksllvn @jbalboni I made the suggested updates to this. It works a lot better then using JavaScript. Thanks for the suggestion.